src/k8s/pkg/k8sd/features/cilium/ingress_test.go

```diff
@@ -0,0 +1,216 @@
+			networkEnabled: false,
+			statusMsg:      fmt.Sprintf(cilium.IngressDeleteFailedMsgTmpl, fmt.Errorf("failed to disable ingress: %w", applyErr)),
+			statusEnabled:  false,
+			helmErr:        applyErr,
+		},
+		{
+			name:           "HelmUnchangedIngressEnabled",
+			ingressEnabled: true,
+			statusMsg:      cilium.EnabledMsg,
+			statusEnabled:  true,
+		},
+		{
+			name:           "HelmUnchangedIngressDisabled",
+			ingressEnabled: false,
+			statusMsg:      cilium.DisabledMsg,
+			statusEnabled:  false,
+		},
+		{
+			name:           "HelmChangedIngressDisabled",
+			applyChanged:   true,
+			ingressEnabled: false,
+			statusMsg:      cilium.DisabledMsg,
+			statusEnabled:  false,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			helmM := &helmmock.Mock{
+				ApplyErr:     tc.helmErr,
+				ApplyChanged: tc.applyChanged,
+			}
+			snapM := &snapmock.Snap{
+				Mock: snapmock.Mock{
+					HelmClient: helmM,
+				},
+			}
+			network := types.Network{
+				Enabled: ptr.To(tc.networkEnabled),
+			}
+			ingress := types.Ingress{
+				Enabled: ptr.To(tc.ingressEnabled),
+			}
+
+			status, err := cilium.ApplyIngress(context.Background(), snapM, ingress, network, nil)
+
+			if tc.helmErr == nil {
+				g.Expect(err).To(BeNil())
+			} else {
+				g.Expect(err).To(MatchError(applyErr))
+			}
+			g.Expect(status.Enabled).To(Equal(tc.statusEnabled))
+			g.Expect(status.Message).To(Equal(tc.statusMsg))
+			g.Expect(status.Version).To(Equal(cilium.CiliumAgentImageTag))
+			g.Expect(helmM.ApplyCalledWith).To(HaveLen(1))
+
+			callArgs := helmM.ApplyCalledWith[0]
+			g.Expect(callArgs.Chart).To(Equal(cilium.ChartCilium))
+			validateIngressValues(g, callArgs.Values, ingress)
+		})
+	}
+}
+
+func TestIngressRollout(t *testing.T) {
+	t.Run("Error", func(t *testing.T) {
+		g := NewWithT(t)
+
+		helmM := &helmmock.Mock{
+			ApplyChanged: true,
+		}
+		clientset := fake.NewSimpleClientset(
+			&v1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "dummy",
+					Namespace: "kube-system",
+				},
+			},
+		)
+		snapM := &snapmock.Snap{
+			Mock: snapmock.Mock{
+				HelmClient: helmM,
+				KubernetesClient: &kubernetes.Client{
+					Interface: clientset,
+				},
+			},
+		}
+		network := types.Network{}
+		ingress := types.Ingress{
+			Enabled: ptr.To(true),
+		}
+
+		status, err := cilium.ApplyIngress(context.Background(), snapM, ingress, network, nil)
+
+		g.Expect(err).To(HaveOccurred())
+		g.Expect(status.Enabled).To(BeFalse())
+		g.Expect(status.Message).To(Equal(fmt.Sprintf(cilium.IngressDeployFailedMsgTmpl,
+			fmt.Errorf("failed to rollout restart cilium to apply ingress: %v",
+				fmt.Errorf("failed to restart cilium-operator deployment after 3 attempts: %w",
+					fmt.Errorf("failed to restart cilium-operator deployment: %w",
+						fmt.Errorf("failed to get deployment cilium-operator in namespace kube-system: %w",
+							errors.New("deployments.apps \"cilium-operator\" not found"))))),
```

Comment:
This is too tightly coupled to the message structure and contents. I see several instances of testing the specific error message in this PR, can those be reverted? I would much rather use a simple errors.Is, if possible.

Reply:
I misunderstood other review comments. I updated the test. Regarding errors when it makes sense i use gomega expect(err).To(ErrorMatch()) which checks if the error is in the chain of errors will that be enough?